packages/hash/api/src/model/knowledge/linkEntity.model.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  GraphApi,
+  Entity,
+  LinkEntityMetadata,
+  Filter,
+  EntityStructuralQuery,
+} from "@hashintel/hash-graph-client";
+
+import { EntityModel, EntityTypeModel, LinkEntityModel } from "../index";
+
+export type LinkModelConstructorParams = {
+  linkEntity: Entity;
+  linkEntityTypeModel: EntityTypeModel;
+  leftEntityModel: EntityModel;
+  rightEntityModel: EntityModel;
+};
+
+export type LinkModelCreateParams = {
+  ownedById: string;
+  linkEntityTypeModel: EntityTypeModel;
+  leftEntityModel: EntityModel;
+  leftOrder?: number;
+  rightEntityModel: EntityModel;
+  rightOrder?: number;
+  actorId: string;
+};
+
+/**
+ * @class {@link LinkEntityModel}
+ */
+export default class extends EntityModel {
+  leftEntityModel: EntityModel;
+
+  rightEntityModel: EntityModel;
+
+  get linkMetadata(): LinkEntityMetadata {
+    return this.entity.metadata.linkMetadata!;
+  }
+
+  constructor({
```

Comment:
code style nit, shouldn't this be above the get method above?

Reply:
Gotcha, so you're putting it above the constructor because that's where you'd define the variable usually?

I _think_ that makes sense, but it adds quite a lot of boiler-plate above the constructor which goes against my intuition. I've tried looking up examples of this to see if there's convention but I can't find any so it's probably unimportant.

My only concern is that the ordering _might_ matter (use-before-define kind of ordering), and I'm not sure if this is a pattern we want to establish as it could maybe cause problems later on if we start mutating inners through these methods or doing things like `self.X = `. I appreciate that won't be a concern here, but just talking about the pattern, and if I was erring on the side of caution I'd probably put them below.

Just some thoughts though, not a request for a change, feel free to leave as is or change if you want :)

Reply:
I think this is not going to be relevant once we remove the getter methods in follow-up